consumers/google.go

```diff
@@ -172,31 +181,40 @@ func (d *googleDNSConsumer) applyChange(change *dns.Change) error {
 	return nil
 }
 
-func (d *googleDNSConsumer) currentRecords() (map[string]*dns.ResourceRecordSet, error) {
+func (d *googleDNSConsumer) currentRecords() (map[string]*ownedRecord, error) {
 	resp, err := d.client.ResourceRecordSets.List(params.project, params.zone).Do()
 	if err != nil {
 		return nil, fmt.Errorf("Error getting DNS records from %s/%s: %v", params.project, params.zone, err)
 	}
 
-	records := make(map[string]*dns.ResourceRecordSet)
+	records := make(map[string]*ownedRecord)
 
 	for _, r := range resp.Rrsets {
-		if r.Type == "TXT" {
-			records[r.Name] = r
-		} else {
-			if _, exists := records[r.Name]; !exists {
-				records[r.Name] = nil
+		if r.Type == "A" || r.Type == "TXT" {
+			record := records[r.Name]
+
+			if record == nil {
```

Comment:
I think it would be a bit more clear if you used the `val, ok := map[key]` pattern here, and check `if !ok` instead of `if record == nil`.